rubi/rubi/contracts/contract_types/transaction_receipt.py

```diff
@@ -47,28 +49,46 @@ def __init__(
         l1_fee_scalar: Optional[Decimal] = None,
         raw_events: Optional[List[BaseEvent]] = None,
     ):
+        # Block number
         self.block_number = block_number
-        self.contract_address = contract_address
-        self.effective_gas_price = effective_gas_price
-        self.gas_used = gas_used
+
+        # Addresses
         self.from_address = from_address
         self.to_address = to_address
-        self.status = status
-        # Do this so that introducing TransactionStatus is not a breaking change
+        self.created_contract_at_address = contract_address
+
+        # Transaction cost
+        self.gas_used = gas_used
+        self.transaction_cost_in_eth = self._calc_gas_cost_in_eth(
```

Comment:
we need to be cautious here, this will only work for `base` and `optimism`, not `arbitrum` as the `TransactionReceipt` differs

Reply:
we should think about how we are going to pass in the relevant `network` to this function, i am not entirely sure but i believe we could possibly be able to discern this information from the `transaction` itself